addons/account/tests/test_tax.py

```diff
@@ -1,1158 +1,1135 @@
+    def fixed_tax(self, amount, **kwargs):
+        self.number += 1
+        return self.env['account.tax'].create({
+            **kwargs,
+            'name': f"fixed_{amount}_({self.number})",
+            'amount_type': 'fixed',
+            'amount': amount,
         })
 
-        cls.bank_journal = cls.company_data['default_journal_bank']
-        cls.bank_account = cls.bank_journal.default_account_id
-        cls.expense_account = cls.company_data['default_account_expense']
-
-    def _check_compute_all_results(self, total_included, total_excluded, taxes, res):
-        self.assertAlmostEqual(res['total_included'], total_included)
-        self.assertAlmostEqual(res['total_excluded'], total_excluded)
-        for i in range(0, len(taxes)):
-            self.assertAlmostEqual(res['taxes'][i]['base'], taxes[i][0])
-            self.assertAlmostEqual(res['taxes'][i]['amount'], taxes[i][1])
+    def _check_compute_all_results(self, taxes, expected_values, price_unit, **kwargs):
+        results = taxes.compute_all(price_unit, **kwargs)
+        self.assertAlmostEqual(results['total_included'], expected_values['total_included'])
+        self.assertAlmostEqual(results['total_excluded'], expected_values['total_excluded'])
+        self.assertEqual(len(results['taxes']), len(expected_values['taxes']))
+        for values, expected_values in zip(results['taxes'], expected_values['taxes']):
+            self.assertEqual(round(values['base'], 6), expected_values[0])
+            self.assertEqual(round(values['amount'], 6), expected_values[1])
 
+    def _assert_tax_results(self, results, expected_values, rounding):
+        """
+        [!] Mirror of the same test in test_account_tax.js.
+        PLZ KEEP BOTH METHODS CONSISTENT WITH EACH OTHERS.
+        """
+        self.assertEqual(
+            float_round(results['total_included'], precision_rounding=rounding),
+            float_round(expected_values['total_included'], precision_rounding=rounding),
+        )
+        self.assertEqual(
+            float_round(results['total_excluded'], precision_rounding=rounding),
+            float_round(expected_values['total_excluded'], precision_rounding=rounding),
+        )
+        self.assertEqual(len(results['tax_values_list']), len(expected_values['tax_values_list']))
+        for tax_values, expected_tax_values in zip(results['tax_values_list'], expected_values['tax_values_list']):
+            self.assertEqual(
+                float_round(tax_values['base'], precision_rounding=rounding),
+                float_round(expected_tax_values[0], precision_rounding=rounding),
+            )
+            self.assertEqual(
+                float_round(tax_values['tax_amount_factorized'], precision_rounding=rounding),
+                float_round(expected_tax_values[1], precision_rounding=rounding),
+            )
+
+    def _check_tax_results(self, taxes, expected_values, price_unit, evaluation_context_kwargs=None, compute_kwargs=None):
```

Comment:
@qdp-odoo I have one mirror with the test js-side that check the tax amounts using the new taxes computation.
The other is checking the results of `compute_all` so a tax amount per repartition line.

Reply:
The first one (_check_tax_results) evaluate the taxes computation and then, call the second one (_assert_tax_results). Do you think it would be cleaner in an inner method?

Reply:
if it's called only by the former and doesn't need to be inherited, yes. Definitively